### PR TITLE
Add StorageDealAwaitingPreCommit state

### PR DESCRIPTION
**Pull request by dirkmc**
Markets side of https://github.com/filecoin-project/lotus/issues/4900

### CODE REVIEW

File: storagemarket/nodes.go [hunk 1]
```diff
@@ -16,6 +16,11 @@ import (
 	"github.com/filecoin-project/go-fil-markets/shared"
 )
 
+// DealSectorCommittedCallback is a callback that runs when a sector is pre-committed
```
Comment (nonsense) [inline storagemarket/nodes.go:19]:
```suggestion
// DealSectorPreCommittedCallback is a callback that runs when a sector is pre-committed
```

**Comment by nonsense** [comment]
I am still very new to this code, but overall looks good to me!

**Review by nonsense — APPROVED** [review]

**Review by hannahhoward — CHANGES_REQUESTED** [review]
It is actually important to move the new deal state to the bottom of the list -- otherwise you need to write a migration for deal states. It's no worries -- I'll do it for you and merge, just a heads up.

**Comment by nonsense** [comment]
> It is actually important to move the new deal state to the bottom of the list -- otherwise you need to write a migration for deal states.

Right, this was going to offset a lot of states, including `Active`, so we would have messed up the state of the client with respect to pretty much all deals. Good catch!

**Review by hannahhoward — APPROVED** [review]

**Pull request merged**